fdbserver/DataDistribution.actor.cpp

```diff
@@ -1762,14 +1949,48 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 
 				state vector<std::vector<UID>> builtTeams;
 
-				int addedTeams = self->addTeamsBestOf(teamsToBuild);
+//				state struct TeamBuildInfo teamBuildInfo;
+//				teamBuildInfo.desiredServerTeamNum = desiredTeams;
+//				teamBuildInfo.maxServerTeamNum = maxTeams;
+
+				int addedTeams = self->addTeamsBestOf(teamsToBuild, desiredTeams, maxTeams);
 				if (addedTeams <= 0 && self->teams.size() == 0) {
 					TraceEvent(SevWarn, "NoTeamAfterBuildTeam")
 					    .detail("TeamNum", self->teams.size())
 					    .detail("Debug", "Check information below");
 					// Debug: set true for traceAllInfo() to print out more information
 					self->traceAllInfo();
 				}
+				if (addedTeams <= 0) {
```

Comment:
Do NOT need to check teamRemover() in the addTeamsBestOf(). You should check teamRemover() when we add an initial team or when a server's location is changed. I..e, check the addTeam() when initialTeam = true, and call teamRemover()